### PR TITLE
#314 - optimization in ModifiableMetadata

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #314 I've made `ModifiableMetadata` merge and clean only if necessary, added corresponding tests.

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/rpm-adapter/issues/329#issuecomment-655685229))

@olenagerasimova Job `gh:artipie/rpm-adapter#329` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/rpm-adapter/issues/329#issuecomment-655685229))

@olenagerasimova Pull request [#329](https://github.com/artipie/rpm-adapter/issues/329) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

File: src/test/java/com/artipie/rpm/pkg/ModifiableMetadataTest.java [hunk 1]
```diff
@@ -95,6 +99,105 @@ void generatesMetadataFile(@TempDir final Path temp) throws IOException {
...
+            )
+        );
+    }
+
+    @Test
+    void generatesMetadataFileWhenRpmsWereAdded(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("filelists.xml");
+        res.toFile().createNewFile();
+        final Path part = temp.resolve("part.filelists.xml");
+        Files.copy(Paths.get("src/test/resources-binary/repodata/filelists.xml.example"), part);
```
Comment (olegmoz) [inline src/test/java/com/artipie/rpm/pkg/ModifiableMetadataTest.java:142]: @olenagerasimova `TestResource` might be used here and in `generatesMetadataFileWhenRpmsWereRemoved` other test as it has been merged

File: src/main/java/com/artipie/rpm/pkg/ModifiableMetadata.java [hunk 2]
```diff
@@ -67,11 +70,24 @@ public ModifiableMetadata(final Metadata origin, final PrecedingMetadata precedi
...
+                    previous, this.origin.output().file(), StandardCopyOption.REPLACE_EXISTING
+                );
+            }
+            final long count;
+            if (pkgs.isEmpty()) {
+                count = this.cnt.get() + new PackagesCount(previous).value();
+            } else {
+                count = this.origin.output().maid().clean(pkgs);
+            }
+            this.cnt.set(count);
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/rpm/pkg/ModifiableMetadata.java]: @olegmoz thanks for noticing, this class is not thread-safe and actually there is no point in using `AtomicLong` here, it's only confusing

**Pull request merged**